sdk/keyvault/azure-security-keyvault-jca/pom.xml

```diff
@@ -33,6 +33,7 @@
       <groupId>org.bouncycastle</groupId>
       <artifactId>bcpkix-lts8on</artifactId>
       <version>2.73.6</version> <!-- {x-version-update;org.bouncycastle:bcpkix-lts8on;external_dependency} -->
+      <optional>true</optional>
```

Comment:
Why do we have this dependency as optional? If we are shading the jar, we don't need this dependency.

Reply:
Currently, we will use the minimum bouncycastle dependency set as shaded source to the fixed directory(`com.azure.security.keyvault.jca.implementation.shaded.org.bouncycastle`), so it's required to as an external dependency.

Reply:
I found similar shading usage, like `gson` in `nimbus-jose-jwt`, see [more](https://bitbucket.org/connect2id/nimbus-jose-jwt/src/94a2e4d11ce322966d30a551a2d482936b588c87/pom.xml#lines-61).
![image](https://github.com/user-attachments/assets/53028de6-d1f7-4861-8fb5-632c7adc9081)

The source code is still use origin path from gson, see [more](https://bitbucket.org/connect2id/nimbus-jose-jwt/src/94a2e4d11ce322966d30a551a2d482936b588c87/src/main/java/com/nimbusds/jose/util/JSONObjectUtils.java#lines-21).
![image](https://github.com/user-attachments/assets/54591309-57d0-4bc4-b91d-c13bcb49e624)